src/plugins/formulas/engine/register.js

```diff
@@ -4,7 +4,37 @@ import { warn } from '../../../helpers/console';
 import { PLUGIN_KEY } from '../formulas';
 import { DEFAULT_LICENSE_KEY, getEngineSettingsWithDefaultsAndOverrides } from './settings';
 
-const ENGINE_KEY = 'engine';
+/**
+ * Prepares and returns the collection for the engine relationship with the HoT instances.
+ *
+ * @returns {Map}
+ */
+function getEngineRelationshipRegistry() {
+  const registryKey = 'engine_relationship';
+  const pluginStaticRegistry = staticRegister(PLUGIN_KEY);
+
+  if (!pluginStaticRegistry.hasItem(registryKey)) {
+    pluginStaticRegistry.register(registryKey, new Map());
+  }
+
+  return pluginStaticRegistry.getItem(registryKey);
+}
+
+/**
+ * Prepares and returns the collection for the engine shared usage.
+ *
+ * @returns {Map}
+ */
+function getSharedEngineUsageRegistry() {
+  const registryKey = 'shared_engine_usage';
+  const pluginStaticRegistry = staticRegister(PLUGIN_KEY);
+
+  if (!pluginStaticRegistry.hasItem(registryKey)) {
+    pluginStaticRegistry.register(registryKey, new Map());
+  }
+
+  return pluginStaticRegistry.getItem(registryKey);
+}
```

Comment:
Looks like we can deduplicate the code for `getEngineRelationshipRegistry` and `getSharedEngineUsageRegistry`
```suggestion
function getSharedEngineUsageRegistry() {
  return getEngineUsageRegistry('shared_engine_usage');
}

function getEngineUsageRegistry(registryKey) {
  const pluginStaticRegistry = staticRegister(PLUGIN_KEY);

  if (!pluginStaticRegistry.hasItem(registryKey)) {
    pluginStaticRegistry.register(registryKey, new Map());
  }

  return pluginStaticRegistry.getItem(registryKey);
}
```